### PR TITLE
Add `holmes generate alertmanager-tests` command

### PR DESCRIPTION
**Pull request by aantn**
and add flags to `holmes investigate alertmanager` to use that data instead of connecting to a live alertmanager

### CODE REVIEW

File: holmes.py [hunk 1]
```diff
@@ -267,6 +280,37 @@ def alertmanager(
     if json_output_file:
         write_json_file(json_output_file, results)
 
+
+@generate_app.command("alertmanager-tests")
+def generate_alertmanager_tests(
+    alertmanager_url: Optional[str] = typer.Option(None, help="AlertManager url"),
```
Comment (Avi-Robusta) [inline holmes.py]: correct me if im wrong but it looks like for this command alertmanager_url is not optional

Comment (aantn) [reply holmes.py]: It's still optional here, because it could be defined in the config file.

**Comment by Avi-Robusta** [comment]
for tests
`poetry run python3 ./holmes.py investigate alertmanager --alertmanager-file ...
`when no url is specified you will get a pydantic exception 
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for Issue                                                                                                                                                                                                     
source_instance_id                                 
```    
since it uses the url and its none in the alertmanager source
I would change the line in `AlertManagerSource` for `source_instance_id` to 

```
        return [
            Issue(
                id=alert.unique_id,
                name=alert.name,
                source_type="prometheus",
                source_instance_id= self.filepath if self.filepath else self.url,
```

**Review by Avi-Robusta — COMMENTED** [review]
LGTM,
2 changes requested

File: holmes.py (hunk 1 above)
Comment (Avi-Robusta) [reply holmes.py]: than specifically in this function I would check before the line
`    source = config.create_alertmanager_source()
`to make the error output shorter for users
This is what I see:
Screen Shot 2024-06-23 at 11 34 03

**Review by Avi-Robusta — APPROVED** [review]

File: holmes.py (hunk 1 above)
Comment (Avi-Robusta) [reply holmes.py]: also a different issue here, want to make sure you dont miss it
https://github.com/robusta-dev/holmesgpt/pull/56#issuecomment-2184900236

Comment (aantn) [reply holmes.py]: I think it's ok here. The last line of output clearly says `--alertmanager-url must be specified`.
If someone misses it or is confused I'll rework it later.

**Comment by aantn** [comment]
> for tests `poetry run python3 ./holmes.py investigate alertmanager --alertmanager-file ... `when no url is specified you will get a pydantic exception
> 
> ```
> pydantic_core._pydantic_core.ValidationError: 1 validation error for Issue                                                                                                                                                                                                     
> source_instance_id                                 
> ```
> 
> since it uses the url and its none in the alertmanager source I would change the line in `AlertManagerSource` for `source_instance_id` to
> 
> ```
>         return [
>             Issue(
>                 id=alert.unique_id,
>                 name=alert.name,
>                 source_type="prometheus",
>                 source_instance_id= self.filepath if self.filepath else self.url,
> ```

Good catch, will fix.

**Review by Avi-Robusta — APPROVED** [review]

**Pull request merged**